pkg/operator/defaultscccontroller/condition.go

```diff
@@ -0,0 +1,84 @@
+package defaultscccontroller
+
+import (
+	"fmt"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// OperatorStatusUpdater is responsible for updating the status block of the operator.
+type OperatorStatusUpdater func(condition operatorv1.OperatorCondition) error
```

Comment:
nit: it updates a condition, not the status. Misnamed?

Reply:
side effects as promise 😱

Reply:
yes, the intent and scope should e clear.